AIM/AIM/Data/Gateways/XMLOnAirInfoGateway.h

```diff
@@ -0,0 +1,30 @@
+//
+//  XMLOnAirInfoGateway.h
+//  AIM
+//
+//  Created by Oskar Szydlowski on 06.03.2018.
+//  Copyright © 2018 Oskar Szydlowski. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+#import "OnAirInfoGateway.h"
+#import "OnAirInfo.h"
+#import "StationInfo.h"
+#import "Track.h"
+
+@interface XMLOnAirInfoGateway : NSObject <NSXMLParserDelegate, OnAirInfoGateway> {}
+
+@property (nonatomic, weak) OnAirInfoCompletion onAirInfoCompletion;
```

Comment:
Are you sure about keeping only weak reference to that completion? This code will work when you will keep strong reference in some other place, but in my opinion it's very fragile condition.